### PR TITLE
Added system to check if a block is flammable on right click of flint and steel (and fixed the whitespace errors)

### PR DESCRIPTION
**Pull request by ghost**
This time I have tested it and it works. There was a TODO in the ItemFlintAndSteel.java and I finished it. I submitted a pull request earlier, but I couldn't build it because I was still messing around, setting up maven and apparently, this I didn't know, maven is sensitive to whitespace... interesting.

### CODE REVIEW

File: src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java [hunk 1]
```diff
@@ -19,17 +19,16 @@ public ItemFlintAndSteel() {
...
-            default:
-                return setBlockOnFire(player, target, face, holding, clickedLoc);
-        }
+        if (target.getType() == Material.OBSIDIAN) {
+            fireNetherPortal();
+        } 
+        if (target.getType() == Material.TNT) {
+            fireTnt(target);
+        } 
+            if (target.isFlammable()) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: Could you go one indent to the left on this line?

**Comment by mastercoms** [comment]
Now that I think about it, I think the correct check would be isOcculuding, since isFlammable just checks if fire can spread to the block. You can still light stone on fire with flint and steel.

**Comment by codekilla747** [comment]
@mastercoms So do I just submit a new pull request when I'm done?

**Comment by twizmwazin** [comment]
@codekilla747 Just push another commit to this branch

**Comment by codekilla747** [comment]
@masteroms How would I implement IsSolid(). Could you point me to the file that has it?

**Comment by codekilla747** [comment]
isSolid() also returns true on sand, brewing stands, barrier blocks, iron doors, iron trapdoors, command blocks, redstone lamps, iron fence (That Exists?), cake blocks, pressure plates... and other blocks that should not be caught on fire. I can add stone to the logic, but I'm not going to use .isSolid().

**Comment by mastercoms** [comment]
I guess we're going to have to add a new method, then.

**Comment by codekilla747** [comment]
@mastercoms Alright, I'll try it tomorrow.

**Comment by codekilla747** [comment]
Setup.sh isn't completing for some reason. I'll push a new commit when i can

File: src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java (hunk 1 above)
Comment (codekilla747) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: @mastercoms fixed in my next commit

**Comment by codekilla747** [comment]
@mastercoms should sand and gravel be able to be ignited?

**Comment by TekkitCommando** [comment]
@codekilla747 Sand and Gravel can be ignited with flint and steel

**Comment by codekilla747** [comment]
This build passed on my local repository so it should pass on CircleCi. I updated what blocks should be ignited. Now maybe it can be merged... @mastercoms

**Comment by codekilla747** [comment]
alright, next commit will be ready to merge

**Comment by mastercoms** [comment]
And finally, I think it would work better as an else if chain, since obsidian and tnt will both trigger twice as of now (resulting in tnt being lit on fire)

**Comment by codekilla747** [comment]
Alright, *now* it should be ready to merge

**Comment by codekilla747** [comment]
Oops, whitespace stuff again

File: src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java [hunk 2]
```diff
@@ -19,17 +19,19 @@ public ItemFlintAndSteel() {
...
+            fireNetherPortal();
+            return true;
+        } 
+        if (target.getType() == Material.TNT) {
+            fireTnt(target);
+            return true;
+        } 
+        if (target.isFlammable() || target.getType().isOccluding()) {
+            setBlockOnFire(player, target, face, holding, clickedLoc);
+            return true
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: Missing semicolon.

Comment (codekilla747) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: O

File: src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java [hunk 3]
```diff
@@ -19,17 +19,19 @@ public ItemFlintAndSteel() {
...
+        } 
+            return false;
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemFlintAndSteel.java]: Tab this over to the left, please.

**Comment by codekilla747** [comment]
@mastercoms Before I make this commit is there anything else you see wrong?

**Comment by mastercoms** [comment]
No, it should be ready to merge after this (yay!)

**Comment by codekilla747** [comment]
Alright, here it goes...

**Comment by codekilla747** [comment]
Alright, now I need to go to school...

**Comment by codekilla747** [comment]
Yay!

**Comment by mastercoms** [comment]
Thank you for your contribution.

**Pull request merged**